File: C1.UWP.FlexChart/VB/GestureChartSample/ViewModel/GestureChartDemoModel.vb
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Linq
Imports System
Imports C1.Xaml.Chart.Interaction

Class GestureChartDemoModel
    Implements INotifyPropertyChanged
    Private _data As List(Of DataPoint)
    Private _currentPointsCount As Integer = 1000

    Public Property Data() As List(Of DataPoint)
        Get
            If _data Is Nothing Then
                _data = DataCreator.Create(_currentPointsCount)
            End If
            Return _data
        End Get
        Set
            _data = Value
            OnPropertyChanged("Data")
        End Set
    End Property

    Public ReadOnly Property GestureMode() As List(Of String)
        Get
            Return [Enum].GetNames(GetType(GestureMode)).ToList()
        End Get
    End Property

    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    Private Sub OnPropertyChanged(propertyName As String)
        RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
    End Sub
End Class

File: C1.UWP.FlexChart/CS/GestureChartSample/ViewModel/GestureChartDemoModel.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel;
using C1.Xaml.Chart.Interaction;

namespace GestureChartSample
{
    class GestureChartDemoModel : INotifyPropertyChanged
    {
        List<DataPoint> _data;
        int _currentPointsCount = 1000;

        public List<DataPoint> Data
        {
            get
            {
                if (_data == null)
                {
                    _data = DataCreator.Create(_currentPointsCount);
                }
                return _data;
            }
            set
            {
                _data = value;
                OnPropertyChanged("Data");
            }
        }

        public List<string> GestureMode
        {
            get
            {
                return Enum.GetNames(typeof(GestureMode)).ToList();
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;

        private void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}
